### PR TITLE
Add explicit type conversion to narrower data type

### PR DESCRIPTION
**Pull request by ThomsonTan**
Add explicit type conversion and more fixes to consistency of using `size_t` and `int`.

### CODE REVIEW

File: sdk/include/opentelemetry/sdk/metrics/aggregator/sketch_aggregator.h [hunk 1]
```diff
@@ -72,7 +72,7 @@ class SketchAggregator final : public Aggregator<T>
     }
     else
     {
-      idx = ceil(log(val) / log(gamma));
+      idx = static_cast<int>(ceil(log(val) / log(gamma)));
```
Comment (pyohannes) [inline sdk/include/opentelemetry/sdk/metrics/aggregator/sketch_aggregator.h:75]: I'm not too familiar with this part of the code, but shouldn't `idx` be a `size_t`? As we use it as an index below.

Comment (ThomsonTan) [reply sdk/include/opentelemetry/sdk/metrics/aggregator/sketch_aggregator.h:75]: I think `size_t` should be ideal but that's a different issue here. The issue here is `double` is returned from `ceil()`. I'll revisit the index type later.

**Review by pyohannes — APPROVED** [review]

File: sdk/test/trace/probability_sampler_test.cc [hunk 2]
```diff
@@ -166,7 +170,8 @@ TEST(ProbabilitySampler, ProbabilitySamplerOnePercent)
 TEST(ProbabilitySampler, ProbabilitySamplerAll)
 {
   double probability = 1.0;
-  int iterations = 100000, expected_count = iterations * probability;
+  int iterations     = 100000;
```
Comment (maxgolov) [inline sdk/test/trace/probability_sampler_test.cc:173]: My only comment here is that the TCs for the probability sampler are very repetitive, containing the same identical code. We could have moved the body into a template method, use constexpr for the percentage parameter, and shrunk the code x10 times. Less LOC is the best.. Maybe even collapsing all into for each - percentage, run, print result, all into just one test that prints a failure if either of the subtests fail.

**Review by maxgolov — APPROVED** [review]

File: sdk/test/trace/probability_sampler_test.cc (hunk 2 above)
Comment (ThomsonTan) [reply sdk/test/trace/probability_sampler_test.cc:173]: This is good suggestion, opened issue https://github.com/open-telemetry/opentelemetry-cpp/issues/350 for tracking.

**Pull request merged**